### PR TITLE
Add FIFO clarification in PriorityQueue docs

### LINKED ISSUE #7397: Please add clarification that PriorityQueue does not guarantee Dequeue order of items with the same priority
[PriorityQueue<TElement,TPriority>](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.priorityqueue-2?view=net-6.0) should clarify that items with the same priority does not have any dequeuing order. [Doc page code is here](https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.Collections.Generic/PriorityQueue%602.xml).

Sorry the name `Queue` made me think it should work like a queue with items with the same priority. Still a clarification should help (and I think it's important?)

Example:

```
var queue = new PriorityQueue<string, int>();
queue.Enqueue("A", 1);
queue.Enqueue("B", -1);
queue.Enqueue("C", 1);

while (queue.TryDequeue(out var foo, out _))
{
    Console.WriteLine(foo);
}
```

Output is
```
B
C
A
```

### PR DESCRIPTION
**Pull request by eiriktsarpalis**
Fix #7397.

### CODE REVIEW

File: xml/System.Collections.Generic/PriorityQueue`2.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
 ## Remarks
 
-Implements an array-backed, quaternary min-heap. Each element is enqueued with an associated priority that determines the dequeue order. Elements with the lowest priority are dequeued first.
+Implements an array-backed, quaternary min-heap. Each element is enqueued with an associated priority that determines the dequeue order. Elements with the lowest priority are dequeued first. Note that the type does not guarantee first-in-first-out semantics for elements of equal priority.
```
Comment (gewarren) [inline xml/System.Collections.Generic/PriorityQueue`2.xml:29]:
```suggestion
Implements an array-backed, quaternary min-heap. Each element is enqueued with an associated priority that determines the dequeue order. Elements with the lowest priority are dequeued first. This type does not guarantee first-in-first-out semantics for elements of equal priority.
```

**Review by gewarren — APPROVED** [review]
Our grammar checker tells us to avoid saying "Note that...".

File: xml/System.Collections.Generic/PriorityQueue`2.xml (hunk 1 above)
Comment (eiriktsarpalis) [reply xml/System.Collections.Generic/PriorityQueue`2.xml:29]: My PR got auto-merged, so your suggestion ended up getting ignored :-(

Comment (gewarren) [reply xml/System.Collections.Generic/PriorityQueue`2.xml:29]: Yes, I wish GitHub didn't automerge if there are suggestions. I don't think it's important enough to create a new PR though.

**Pull request merged**